# Task
Analyze the 'teen_phone_addiction_processed.xlsx' dataset by examining the 'Addiction_Level' column, identifying other numerical features, and preparing for pairwise statistical comparisons between these features and 'Addiction_Level' groups.

## Load the dataset

### Subtask:
Read the 'teen_phone_addiction_processed.xlsx' file into a pandas DataFrame.


**Reasoning**:
To load the dataset, I will import pandas, read the Excel file, display the first few rows, and show the column names and data types to verify the data structure.



In [ ]:
import seaborn as sns
import matplotlib.pyplot as plt

# Select all numerical columns for pairwise correlation, excluding 'ID'
numerical_cols_for_heatmap = [col for col in df.select_dtypes(include=['float64', 'int64']).columns if col != 'ID']

# Calculate the Pearson correlation matrix
correlation_matrix_full = df[numerical_cols_for_heatmap].corr(method='pearson')

# Plot the heatmap
plt.figure(figsize=(20, 18)) # Adjust figure size for better readability
sns.heatmap(correlation_matrix_full, annot=True, cmap='coolwarm', fmt=".2f", linewidths=.5)
plt.title('Pairwise Pearson Correlation Heatmap of Numerical Features')
plt.xticks(rotation=90)
plt.yticks(rotation=0)
plt.tight_layout()
plt.show()

NameError: name 'df' is not defined

In [ ]:
import pandas as pd

# Read the Excel file into a pandas DataFrame
df = pd.read_excel('/content/teen_phone_addiction_processed.xlsx')

# Display the first few rows of the DataFrame
print("First 5 rows of the DataFrame:")
print(df.head())

# Display the column names and their data types
print("\nColumn names and their data types:")
print(df.info())

First 5 rows of the DataFrame:
   ID  Age  Daily_Usage_Hours  Sleep_Hours  Academic_Performance  \
0   1   13                4.0          6.1                    78   
1   2   17                5.5          6.5                    70   
2   3   13                5.8          5.5                    93   
3   4   18                3.1          3.9                    78   
4   5   14                2.5          6.7                    56   

   Social_Interactions  Exercise_Hours  Anxiety_Level  Depression_Level  \
0                    5             0.1             10                 3   
1                    5             0.0              3                 7   
2                    8             0.8              2                 3   
3                    8             1.6              9                10   
4                    4             1.1              1                 5   

   Self_Esteem  ...  School_Grade_11th  School_Grade_12th  School_Grade_7th  \
0            8  ...           

## Explore 'Addiction_Level' column

### Subtask:
Identify the unique categories and their distribution within the 'Addiction_Level' column.


**Reasoning**:
To identify the unique categories and their distribution within the 'Addiction_Level' column, I will use the `value_counts()` method on this column and display the results.



In [ ]:
print("Distribution of 'Addiction_Level' column:")
print(df['Addiction_Level'].value_counts())

Distribution of 'Addiction_Level' column:
Addiction_Level
10.0    1524
9.9       50
9.6       46
8.0       45
7.8       45
        ... 
2.0        1
2.2        1
2.3        1
1.0        1
3.6        1
Name: count, Length: 80, dtype: int64


## Identify other numerical columns

### Subtask:
List other numerical columns in the dataset that can be used for pairwise comparisons against 'Addiction_Level'.


**Reasoning**:
To identify other numerical columns, I will filter the DataFrame's columns by their data types, selecting only 'float64' and 'int64' types, and then exclude 'ID' and 'Addiction_Level' as specified.



In [ ]:
numerical_cols = df.select_dtypes(include=['float64', 'int64']).columns.tolist()

# Exclude 'ID' and 'Addiction_Level' from the list
excluded_cols = ['ID', 'Addiction_Level']
other_numerical_cols = [col for col in numerical_cols if col not in excluded_cols]

print("Numerical columns for pairwise comparisons (excluding 'ID' and 'Addiction_Level'):")
print(other_numerical_cols)

Numerical columns for pairwise comparisons (excluding 'ID' and 'Addiction_Level'):
['Age', 'Daily_Usage_Hours', 'Sleep_Hours', 'Academic_Performance', 'Social_Interactions', 'Exercise_Hours', 'Anxiety_Level', 'Depression_Level', 'Self_Esteem', 'Parental_Control', 'Screen_Time_Before_Bed', 'Phone_Checks_Per_Day', 'Apps_Used_Daily', 'Time_on_Social_Media', 'Time_on_Gaming', 'Time_on_Education', 'Family_Communication', 'Weekend_Usage_Hours', 'Checks_per_Usage_Hour', 'Weekend_Weekday_Diff', 'Share_Social', 'Share_Gaming', 'Share_Education', 'Other_Usage_Hours', 'Late_Screen_Flag', 'Mental_Health_Risk_Z']


## Prepare for pairwise analysis

### Subtask:
Prepare the 'Addiction_Level' column for statistical grouping and outline the methodology for pairwise statistical comparisons between numerical features and these 'Addiction_Level' groups.


## Summary:

### Data Analysis Key Findings
*   The `teen_phone_addiction_processed.xlsx` dataset was successfully loaded, comprising 3000 rows and 42 columns, with no missing data. The column types include 14 boolean, 16 float64, and 12 int64 columns.
*   The `Addiction_Level` column contains 80 unique values, with `10.0` being the most frequent, appearing 1524 times. Many other `Addiction_Level` values appear infrequently, some only once.
*   A total of 26 numerical columns, such as `Age`, `Daily_Usage_Hours`, `Sleep_Hours`, `Academic_Performance`, `Anxiety_Level`, and `Depression_Level`, were identified for subsequent pairwise statistical comparisons with the `Addiction_Level` column.

### Insights or Next Steps
*   Given the high number of unique values (80) and the heavily skewed distribution of the `Addiction_Level` column (over 50\% of records are `10.0`), directly using these values for group comparisons might be statistically challenging.
*   The next step should focus on grouping the `Addiction_Level` into a more manageable number of categories (e.g., "Low Addiction", "Medium Addiction", "High Addiction") to facilitate robust statistical analysis and meaningful pairwise comparisons with other numerical features.


In [ ]:
import numpy as np

# Define the bins and labels for the new Addiction_Category column
bins = [0, 3.5, 6.5, np.inf]  # Using 0 as the lower bound and np.inf for the upper bound
labels = ['Low Addiction', 'Medium Addiction', 'High Addiction']

# Create the new categorical column
df['Addiction_Category'] = pd.cut(
    df['Addiction_Level'],
    bins=bins,
    labels=labels,
    right=False  # This means the bins are inclusive on the left, exclusive on the right
)

# Display the distribution of the new categorical column
print("\nDistribution of 'Addiction_Category' column:")
print(df['Addiction_Category'].value_counts())

# Display the first few rows with the new column to verify
print("\nFirst 5 rows with 'Addiction_Category':")
print(df[['Addiction_Level', 'Addiction_Category']].head())


Distribution of 'Addiction_Category' column:
Addiction_Category
High Addiction      2689
Medium Addiction     288
Low Addiction         23
Name: count, dtype: int64

First 5 rows with 'Addiction_Category':
   Addiction_Level Addiction_Category
0             10.0     High Addiction
1             10.0     High Addiction
2              9.2     High Addiction
3              9.8     High Addiction
4              8.6     High Addiction


### One-way ANOVA for Academic_Performance vs Addiction_Category

To determine if there are statistically significant differences in the mean 'Academic_Performance' across the 'Low Addiction', 'Medium Addiction', and 'High Addiction' categories, we will perform a one-way ANOVA test.

**Hypotheses:**
*   **Null Hypothesis (H0):** The mean academic performance is the same across all 'Addiction_Category' groups.
*   **Alternative Hypothesis (H1):** At least one 'Addiction_Category' group has a different mean academic performance.

In [ ]:
from scipy.stats import f_oneway

# Extract 'Academic_Performance' for each addiction category
low_addiction_academic_performance = df[df['Addiction_Category'] == 'Low Addiction']['Academic_Performance']
medium_addiction_academic_performance = df[df['Addiction_Category'] == 'Medium Addiction']['Academic_Performance']
high_addiction_academic_performance = df[df['Addiction_Category'] == 'High Addiction']['Academic_Performance']

# Perform one-way ANOVA
f_statistic, p_value = f_oneway(low_addiction_academic_performance, medium_addiction_academic_performance, high_addiction_academic_performance)

print(f"One-way ANOVA for Academic_Performance vs Addiction_Category:")
print(f"F-statistic: {f_statistic:.4f}")
print(f"P-value: {p_value:.4f}")

# Interpret the results
alpha = 0.05
if p_value < alpha:
    print("\nSince the p-value is less than the significance level (0.05), we reject the null hypothesis.")
    print("This suggests that there are significant differences in the mean academic performance across the addiction categories.")
else:
    print("\nSince the p-value is greater than the significance level (0.05), we fail to reject the null hypothesis.")
    print("This suggests that there are no significant differences in the mean academic performance across the addiction categories.")

One-way ANOVA for Academic_Performance vs Addiction_Category:
F-statistic: 0.7066
P-value: 0.4934

Since the p-value is greater than the significance level (0.05), we fail to reject the null hypothesis.
This suggests that there are no significant differences in the mean academic performance across the addiction categories.


In [ ]:
import pandas as pd

# Calculate Pearson correlation between 'Addiction_Level' and other numerical columns
# Ensure 'Addiction_Level' is included in the DataFrame for correlation calculation
correlation_data = df[['Addiction_Level'] + other_numerical_cols]

# Calculate the correlation matrix
correlation_matrix = correlation_data.corr(method='pearson')

# Extract the correlations specifically with 'Addiction_Level'
addiction_level_correlations = correlation_matrix['Addiction_Level'].drop('Addiction_Level')

print("Pearson Correlation with 'Addiction_Level':")
print(addiction_level_correlations.sort_values(ascending=False))

# Interpret the correlations
print("\nInterpretation of Pearson's Correlation Coefficients:")
print("-   Values close to 1 indicate a strong positive linear relationship.")
print("-   Values close to -1 indicate a strong negative linear relationship.")
print("-   Values close to 0 indicate a weak or no linear relationship.")

Pearson Correlation with 'Addiction_Level':
Daily_Usage_Hours         0.600771
Apps_Used_Daily           0.319287
Time_on_Social_Media      0.306578
Time_on_Gaming            0.273060
Phone_Checks_Per_Day      0.246342
Other_Usage_Hours         0.198795
Late_Screen_Flag          0.036133
Age                       0.031306
Mental_Health_Risk_Z      0.026649
Anxiety_Level             0.016005
Screen_Time_Before_Bed    0.013784
Academic_Performance      0.012264
Depression_Level          0.008491
Time_on_Education        -0.000586
Parental_Control         -0.001016
Family_Communication     -0.010482
Social_Interactions      -0.010631
Weekend_Usage_Hours      -0.013049
Exercise_Hours           -0.021015
Self_Esteem              -0.022292
Share_Gaming             -0.071088
Share_Social             -0.152751
Checks_per_Usage_Hour    -0.199194
Sleep_Hours              -0.216681
Share_Education          -0.340989
Weekend_Weekday_Diff     -0.432135
Name: Addiction_Level, dtype: float64

Interpr

I will now calculate the Pearson correlation coefficient between 'Addiction_Level' and all other numerical columns. This will show us how strongly and in what direction each numerical feature is linearly related to the addiction level.

The Pearson Correlation with 'Addiction_Level' has been calculated for all numerical features. Here's what we found:

Strong Positive Correlations (as Addiction_Level increases, so do these):

Daily_Usage_Hours: 0.6008 (Moderately strong positive correlation)
Apps_Used_Daily: 0.3193
Time_on_Social_Media: 0.3066
Time_on_Gaming: 0.2731
Phone_Checks_Per_Day: 0.2463
Strong Negative Correlations (as Addiction_Level increases, these decrease):

Weekend_Weekday_Diff: -0.4321 (Moderately strong negative correlation)
Share_Education: -0.3410
Sleep_Hours: -0.2167
Checks_per_Usage_Hour: -0.1992
Weak or No Linear Correlation (values close to 0): Many features fall into this category, including Academic_Performance (0.0123), Age (0.0313), Anxiety_Level (0.0160), Depression_Level (0.0085), Mental_Health_Risk_Z (0.0266), Parental_Control (-0.0010), Family_Communication (-0.0105), and Social_Interactions (-0.0106).

In summary: Features like daily phone usage, app usage, and time spent on social media and gaming show a positive linear relationship with 'Addiction_Level'. Conversely, differences in weekend vs. weekday usage, the share of time spent on education, and sleep hours show a negative linear relationship. Notably, Academic_Performance shows a very weak positive correlation, consistent with our earlier ANOVA results that found no significant mean differences across addiction categories.

### Kruskal-Wallis H-test for Academic_Performance vs Addiction_Category

Since the assumptions for ANOVA might not be met, or as an alternative test, we will perform a non-parametric Kruskal-Wallis H-test to determine if there are statistically significant differences in the median 'Academic_Performance' across the 'Low Addiction', 'Medium Addiction', and 'High Addiction' categories.

**Hypotheses:**
*   **Null Hypothesis (H0):** The median academic performance is the same across all 'Addiction_Category' groups.
*   **Alternative Hypothesis (H1):** At least one 'Addiction_Category' group has a different median academic performance.

In [ ]:
from scipy.stats import kruskal

# Perform Kruskal-Wallis H-test
h_statistic, p_value_kruskal = kruskal(low_addiction_academic_performance,
                                       medium_addiction_academic_performance,
                                       high_addiction_academic_performance)

print(f"Kruskal-Wallis H-test for Academic_Performance vs Addiction_Category:")
print(f"H-statistic: {h_statistic:.4f}")
print(f"P-value: {p_value_kruskal:.4f}")

# Interpret the results
alpha = 0.05
if p_value_kruskal < alpha:
    print("\nSince the p-value is less than the significance level (0.05), we reject the null hypothesis.")
    print("This suggests that there are significant differences in the median academic performance across the addiction categories.")
    print("Proceeding with Dunn's post-hoc test.")
else:
    print("\nSince the p-value is greater than the significance level (0.05), we fail to reject the null hypothesis.")
    print("This suggests that there are no significant differences in the median academic performance across the addiction categories.")
    print("Dunn's post-hoc test is not necessary as there's no overall significant difference.")

Kruskal-Wallis H-test for Academic_Performance vs Addiction_Category:
H-statistic: 1.4296
P-value: 0.4893

Since the p-value is greater than the significance level (0.05), we fail to reject the null hypothesis.
This suggests that there are no significant differences in the median academic performance across the addiction categories.
Dunn's post-hoc test is not necessary as there's no overall significant difference.


### Dunn's Post-Hoc Test

If the Kruskal-Wallis H-test yields a significant p-value, we proceed with Dunn's post-hoc test to identify which specific pairs of 'Addiction_Category' groups have significantly different median 'Academic_Performance'. This test helps in making pairwise comparisons while controlling for the multiple comparisons problem.

In [ ]:
# Install scikit-posthocs if not already installed
!pip install scikit-posthocs


In [ ]:
import scikit_posthocs as sp

# Prepare data for Dunn's test in the required format (list of arrays/series)
data_for_dunn = [
    df[df['Addiction_Category'] == 'Low Addiction']['Academic_Performance'],
    df[df['Addiction_Category'] == 'Medium Addiction']['Academic_Performance'],
    df[df['Addiction_Category'] == 'High Addiction']['Academic_Performance']
]

# Perform Dunn's test with Bonferroni correction
dunn_results = sp.posthoc_dunn(data_for_dunn, p_adjust='bonferroni')

# Add meaningful row and column labels
dunn_results.columns = ['Low Addiction', 'Medium Addiction', 'High Addiction']
dunn_results.index = ['Low Addiction', 'Medium Addiction', 'High Addiction']

print("Dunn's Post-Hoc Test Results (p-values with Bonferroni correction):")
print(dunn_results)

# Interpret the results (optional, manual interpretation for clarity)
print("\nPairwise comparisons where p-value < 0.05 are statistically significant:")
for i in range(len(dunn_results.index)):
    for j in range(i + 1, len(dunn_results.columns)):
        pair1 = dunn_results.index[i]
        pair2 = dunn_results.columns[j]
        p_val = dunn_results.iloc[i, j]
        if p_val < alpha:
            print(f"  {pair1} vs {pair2}: p = {p_val:.4f} (Significant)")
        else:
            print(f"  {pair1} vs {pair2}: p = {p_val:.4f} (Not Significant)")

Dunn's Post-Hoc Test Results (p-values with Bonferroni correction):
                  Low Addiction  Medium Addiction  High Addiction
Low Addiction               1.0          1.000000        1.000000
Medium Addiction            1.0          1.000000        0.712366
High Addiction              1.0          0.712366        1.000000

Pairwise comparisons where p-value < 0.05 are statistically significant:
  Low Addiction vs Medium Addiction: p = 1.0000 (Not Significant)
  Low Addiction vs High Addiction: p = 1.0000 (Not Significant)
  Medium Addiction vs High Addiction: p = 0.7124 (Not Significant)


The Dunn's post-hoc test results, with Bonferroni correction, indicate no statistically significant differences in 'Academic_Performance' between any of the 'Addiction_Category' groups.

Here are the p-values for each pairwise comparison:

Low Addiction vs Medium Addiction: p = 1.0000
Low Addiction vs High Addiction: p = 1.0000
Medium Addiction vs High Addiction: p = 0.7212
Since all these p-values are greater than our significance level (alpha = 0.05), we conclude that there are no significant differences in the median 'Academic_Performance' when comparing 'Low Addiction' with 'Medium Addiction', 'Low Addiction' with 'High Addiction', or 'Medium Addiction' with 'High Addiction'. This aligns with the earlier Kruskal-Wallis H-test result, which also found no overall significant difference across the groups.